controllers/v1/gardens/gardens_controller.go

```diff
@@ -155,7 +166,24 @@ func GetGardensByUserId(fireBaseUserId interface{}) []gardens_models.Gardens {
 		log.Println(cursorErr)
 	}
 
-	return results
+	gardenCategories, gardenCategoryErr := garden_categories_controllers.GetGardenCategories()
+
+	if gardenCategoryErr != nil {
+		log.Println("Error getting garden categories to populate Gardens by fireBaseUserId")
+	}
+
+	var populatedGardens []gardens_models.Gardens
+	for _, garden := range results {
+		for _, gardenCategory := range gardenCategories {
+			if gardenCategory.ID == garden.GardenCategoryId {
```

Comment:
I guess it would be ok maybe , because theres not hundreds of gardens and categories.
Let test it and if its too much load then let think something else or just let the frontend filter it.(but i think it'd be fine)

Reply:
Yeah, we can test this out. Another way we can do it is call the database multiple times per garden.